lib/iris/fileformats/pp.py

```diff
@@ -1581,51 +1588,68 @@ def _interpret_fields(fields):
     numpy arrays (via the _create_field_data) function.
 
     """
-    land_mask = None
+    land_mask_field = None
     landmask_compressed_fields = []
     for field in fields:
         # Store the first reference to a land mask, and use this as the
         # definitive mask for future fields in this generator.
-        if land_mask is None and field.lbuser[6] == 1 and \
+        if land_mask_field is None and field.lbuser[6] == 1 and \
                 (field.lbuser[3] // 1000) == 0 and \
                 (field.lbuser[3] % 1000) == 30:
-            land_mask = field
+            land_mask_field = field
```

Comment:
@pp-mo We're assuming that the `land_mask_field` is constant, right? i.e. multiple occurrences are simply duplicates?

Reply:
No problem here, discussed to clarify. Always snaps the first land/sea mask.